Add 11ty for docs content

BigBlueHat opened the pull request:
The Markdown and templates (see `_layouts`) live in
the `src/` folder. All other static content (without
any need for processing) live directly in the `content/`
folder. Eleventy will build what is in `src/` and overwrite
the equivalents in `content/`--it will not delete anything,
however.

The `content/` folder should still be committed to the repo
because we serve out of what is on disk immediately after
checkout from the `content/` folder.

BigBlueHat committed:
Add 11ty for docs content

The Markdown and templates (see `_layouts`) live in
the `src/` folder. All other static content (without
any need for processing) live directly in the `content/`
folder. Eleventy will build what is in `src/` and overwrite
the equivalents in `content/`--it will not delete anything,
however.

The `content/` folder should still be committed to the repo
because we serve out of what is on disk immediately after
checkout from the `content/` folder.

BigBlueHat commented:
Not really any content of value, but the setup is in place. 😁

Treora reviewed (commented):
Great progress. Small comments:

- I see a file `content/getting-started/more.html`, which is missing in the `src` folder. Accidental leftover?
- I presume that happens because, as you said “it will not delete anything, however”. Might we want to just run `rm -rf content/docs` before running eleventy? (like we do in `build:demo`)
- Shall we use eleventy for the whole site, not just the docs? E.g. the top-level `content/index.html` now overlaps a lot with `src/_layouts/default.liquid`; would nice to deduplicate.
- In case we keep it only for docs, perhaps `src` is better renamed to e.g. `docs` or `src/docs`.

BigBlueHat committed:
Remove more/index.html cruft

BigBlueHat committed:
Move demo copying code to eleventy config

BigBlueHat committed:
Move static content into src/

We should (now) be able to blow away `content/` and
recreate it from `src/` via eleventy--however, this
commit does not destroy the `content/` folder in
the build process, though a future commit probably
will introduce that.

BigBlueHat commented:
@Treora made a few of your requested changes. 😃 The `index.html` file is still not converted into Markdown, but could/can be at any point--right now 11ty just copies it over.

Feel free to merge this if you're "happy enough." No reason we have to block on the `index.html` conversion (imo), but you're call. ☎️
